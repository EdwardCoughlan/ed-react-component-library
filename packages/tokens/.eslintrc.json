{
  "extends": [
    "plugin:@nrwl/nx/react",
    "../../.eslintrc.json",
    "airbnb",
    "airbnb-typescript"
  ],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "parserOptions": {
        "project": ["packages/tokens/tsconfig.*?.json"]
      },
      "rules": {
        "react/react-in-jsx-scope": "off",
        "react/require-default-props": "off",
        "react/function-component-definition": "off"
      }
    },
    {
      "files": ["*.spec.ts", "*.spec.tsx", "*.stories.*"],
      "rules": {
        "import/no-extraneous-dependencies": "off"
      }
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {
        "@typescript-eslint/explicit-function-return-type": ["error"]
      }
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
