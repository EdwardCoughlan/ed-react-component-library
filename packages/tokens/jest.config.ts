/* eslint-disable */
/* eslint-disable */
/* eslint-disable */
export default {
  displayName: 'tokens',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/packages/tokens',
};
