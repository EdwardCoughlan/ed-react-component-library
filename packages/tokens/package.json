{
  "name": "@edwardcoughlan/tokens",
  "version": "0.0.0",
  "style": "index.css",
  "peerDependencies": {
    "react": "18.1.0",
    "react-dom": "18.1.0",
    "classnames": "^2.3.1"
  },
  "exports":{
    "./index.css": {
      "import" : "./index.css"
    }
  } 
}
