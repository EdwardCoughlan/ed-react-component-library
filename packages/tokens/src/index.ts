export { Tokens } from './lib/tokens';
export type { TokensProps } from './lib/tokens';
export {
  Bloc, BlocProvider, BlocFn, BlocContext,
} from './lib/bloc';
export type { BlocProps, ComponentMapper, ComponentMapperType } from './lib/bloc';
export { useSlots } from './lib/useSlots';
export { RootCss } from './lib/rootCss';
