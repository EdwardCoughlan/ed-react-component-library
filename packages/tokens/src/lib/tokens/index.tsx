export { Tokens } from './tokens';
export type { TokensProps } from './tokens';
