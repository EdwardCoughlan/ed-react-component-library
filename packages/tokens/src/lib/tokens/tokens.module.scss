.background {
  background-color: white;
}

.hovered {
  background-color: red;
}
