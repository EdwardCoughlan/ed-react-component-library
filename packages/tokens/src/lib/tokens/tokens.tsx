import cn from 'classnames';
import { useState } from 'react';
import classes from './tokens.module.scss';

export type TokensProps = {
  welcomeMessage?: string;
};

export const Tokens = ({ welcomeMessage = 'Welcome to Eds Tokens!' }: TokensProps): JSX.Element => {
  const [isHovered, setIsHovered] = useState(false); return (
    <div className={cn(classes.background, isHovered ? classes.hovered : '')} onMouseEnter={(): void => setIsHovered(true)} onMouseLeave={(): void => setIsHovered(false)}>
      {' '}
      <h1>{welcomeMessage}</h1>
    </div>
  );
};
export default Tokens;
