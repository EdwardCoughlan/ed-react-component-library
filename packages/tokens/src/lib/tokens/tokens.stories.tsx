import { ComponentStory, Meta } from '@storybook/react';
import { Tokens, TokensProps } from './tokens';

export default {
  component: Tokens,
  title: 'Tokens',
} as Meta;

const Template: ComponentStory<typeof Tokens> = ({
  welcomeMessage,
}: TokensProps) => <Tokens welcomeMessage={welcomeMessage} />;

export const Primary = Template.bind({});
Primary.args = {
  welcomeMessage: 'hi from storybook',
};
