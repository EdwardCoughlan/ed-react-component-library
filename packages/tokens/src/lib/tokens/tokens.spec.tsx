import { render } from '@testing-library/react';

import { Tokens } from './tokens';

describe('Tokens', () => {
  it('should render successfully', () => {
    const { baseElement, container } = render(<Tokens />);
    expect(baseElement).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
