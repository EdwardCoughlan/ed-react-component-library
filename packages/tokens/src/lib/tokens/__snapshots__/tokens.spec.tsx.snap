// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tokens should render successfully 1`] = `
<div>
  <div
    class="background"
  >
     
    <h1>
      Welcome to Eds Tokens!
    </h1>
  </div>
</div>
`;
